File: swift/perfect-numbers/Sources/PerfectNumbers/PerfectNumbers.swift
enum NumberType {
    case perfect
    case deficient
    case abundant
}

class NumberClassifier {

    public var classification: NumberType

    public init(number: Int) {

        let sum = (1..<number).filter {
            number % $0 == 0
        }.reduce(0, +)

        if sum < number {
            classification = .deficient
        } else if sum > number {
            classification = .abundant
        } else {
            classification = .perfect
        }

    }
}

File: vbnet/perfect-numbers/PerfectNumbers.vb
Imports System
Imports System.Linq

Public Enum Classification
    Perfect
    Abundant
    Deficient
End Enum

Public Module PerfectNumbers
    Public Function Classify(number As Integer) As Classification

        If number < 1
            Throw new ArgumentOutOfRangeException()
        End If

        Dim sum = Enumerable.Range(1, number/2).Where(function(i) number Mod i = 0).Sum()

        If sum > number
            Return Classification.Abundant
        ElseIf sum < number
            Return Classification.Deficient
        Else
            Return Classification.Perfect
        End If
    End Function
End Module
